src/main/java/com/shapesecurity/shift/fuzzer/Fuzzer.java

```diff
@@ -1109,8 +1122,15 @@ private static VariableDeclarationStatement randomVariableDeclarationStatement(@
 
   @NotNull
   private static VariableDeclarator randomVariableDeclarator(@NotNull GenCtx ctx, int depth) {
-    return new VariableDeclarator(randomBinding(ctx, depth - 1), // false, false
-     optional(Fuzzer::randomExpression).apply(ctx, depth - 1));
+    Binding binding;
+    do {
+      binding = randomBinding(ctx, depth - 1);
+    } while (binding instanceof MemberExpression);
```

Comment:
We should try to get rid of these do-whiles and replace them with generators that only generate the appropriate nodes in the first place.